dev_guide/openshift_pipeline.adoc

```diff
@@ -0,0 +1,115 @@
+API from within the Jenkins slaves.  The OpenShift DSL is based on Groovy syntax and provides many convenience
+methods for controlling the lifecycle of your application such as Create, Build, Deploy, and Delete to name a few.
+
+The full details of the API are embedded within the plugin's online documentation within a running Jenkins instance. To find it:
+
+* Create a new Pipeline Item
+* Click "Pipeline Syntax" below the DSL text area
+* On the left navigation menu, click "Global Variables Reference"
+
+
+[[jenkins-pipeline-strategy]]
+
+== Jenkins Pipeline Strategy
+
+In order to take advantage of the OpenShift Pipelines within your project, you will need to utilize the
+xref:../../dev_guide/builds/build_strategies.html#pipeline-strategy-options[Jenkins Pipeline Build Strategy].
+This strategy defaults to using a *Jenkinsfile* at the root of your source repository, but also provides the
+following configuration options:
+
+* an inline jenkinsFile field within your BuildConfig
+* a jenkinsfilePath field within your BuildConfig that references the location of the JenkinsFile to use relative to the source contextDir.
+
+[NOTE]
+====
+The optional *jenkinsfilePath* field specifies the name of the file to use, relative to the source *contextDir*.
+If *contextDir* is omitted, it defaults to the root of the repository. If *jenkinsfilePath* is omitted, it defaults to *Jenkinsfile*.
+====
+
+For more detailed information about the Jenkins Pipeline Strategy see xref:../../dev_guide/builds/build_strategies.html#pipeline-strategy-options[Pipeline Strategy Options].
+
+[[the-jenkinsfile]]
+
+== Jenkinsfile
+
+The Jenkinsfile utilizes the standard groovy language syntax to allow fine grained control
+over the configuration, build, and deployment of your application.
+
+The Jenkinsfile can be supplied in one of the following ways:
+
+- a file located within your source code repository
+- embedded as part of your build configuration using the *jenkinsfile* field
+
+When using the first option, the Jenkinsfile must be included in your applications source code repository at one of the following locations:
+
+- A file named *Jenkinsfile* at the root of your repository
+- A file named *Jenkinsfile* at the root of the source *contextDir* of your repository
+- A file name specified via the *jenkinsfilePath* field of the *JenkinsPiplineStrategy* section of your build config, which is relative to the source *contextDir* if supplied, otherwise it defaults to the root of the repository
+
+The Jenkinsfile is executed on the Jenkins slave pod, which must have the OpenShift Client binaries available if you intend to use the OpenShift DSL.
+
+[[tutorial]]
+
+== Tutorial
+
+For a full walkthrough of building and deploying an application via a Jenkins Pipeline see xref:../app_tutorials/jenkins_pipeline.html#overview[Jenkins Pipeline Tutorial].
+
+[[advanced-topics]]
+
+== Advanced Topics
```

Comment:
If this was previously discussed I missed it ... but I would move the entire Advanced Topics section to configuring_pipeline_execution

Reply:
The point of this section is to get this information into the online documentation and into the hands of developers.  There is similar information already in the install_config section

Reply:
ok seems maybe redundant to me then ... or conversely, if we mention this stuff here, we should mention the sync plugin information here as well (it is similar is scope / feel in my opinion)

Thoughts @coreydaley  @bparees ?

Reply:
the install config section isn't available to users reading the online docs (because it's assumed they aren't doing install/config)
https://docs.openshift.com/online/welcome/index.html

Reply:
(which means referencing that section is also going to be problematic of course.  might have to ask @bmcelvee what's best to do here).

But i think the goal of getting this in front of developers (who would not be reading the install/config docs) is a valid one.

Reply:
The overall goal here is to explain to developers what pipelines are/how they work and reference out to more detailed specifics elsewhere.  so mentioning the sync plugin makes sense (w/ a link to either the jenkins image doc that talks more about it).

Reply:
We can `ifdef` the content so it only appears in Online docs, reducing the redundancy in our OCP and Origin docs. Would that be a good solution?

Reply:
Not great imho, I just don't expect developers to read the install /config docs at all unless we point them there. I'm more concerned that for the online  docs we're pointing to an invalid link.

Writing the full content here for online and linking it for origin could be an option but that seems painful to maintain.